xonsh/foreign_shells.py

```diff
@@ -580,11 +580,13 @@ def load_foreign_aliases(shells=None, config=None, issue_warning=True):
     for shell in shells:
         shell = ensure_shell(shell)
         _, shaliases = foreign_shell_data(**shell)
+        shaliases = {} if shaliases is None else shaliases
         for alias in set(shaliases) & set(xonsh_aliases):
-            del shaliases[alias]
-            print('aliases: alias {!r} of shell {!r} '
-                  'tries to override xonsh alias, '
-                  'xonsh wins!'.format(alias, shell['shell']),
-                  file=sys.stderr)
+            if alias in shaliases:
```

Comment:
I checked out this branch and merged in mine then resolve the conflicts. We can merge this before merging mine. Do you want me to merge/rebase this so it will properly be shown in commit history?